### PR TITLE
Add the new API notebooks

### PR DESCRIPTION
**Pull request by lhagen-isb**

### CODE REVIEW

File: Notebooks/How_to_use_ISB-CGC_APIs.Rmd [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+```{r}
+# Check that the wasn't an error with the request
+if (about_req$status_code != "200") {
+  # Print the error code if something went wrong
+  print(about_req$status_code)
+}
+```
+
+Finally, we will print out the information that we have recieved from the API. This response returns as a response object. You can access different parts of the with helper methods such as `content()` to access the body of the request.
```
Comment (DeenaBleich) [inline Notebooks/How_to_use_ISB-CGC_APIs.Rmd]: 'recieve' should be 'receive'

File: Notebooks/How_to_use_ISB-CGC_APIs.Rmd [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+```{r}
+# First submit the 'get' request to the API
+about_req <- GET("https://api-dot-isb-cgc.appspot.com/v4/about")
+```
+
+Now that we have the request response, we are going to check that we didn't recieve an error code or if the request was successful. If the request was successful, then the status code will come back as 200 but if something went wrong then the status code may be something 404 or 503. If you have recieved any error codes, you can check out Google's [Troubleshooting response errors guide](https://cloud.google.com/endpoints/docs/openapi/troubleshoot-response-errors).
+
+```{r}
+# Check that the wasn't an error with the request
```
Comment (DeenaBleich) [inline Notebooks/How_to_use_ISB-CGC_APIs.Rmd]: Should be "Check that there wasn't..."

File: Notebooks/How_to_use_ISB-CGC_APIs.Rmd [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+about_req
+```
+
+```{r}
+# Next we display the body of the request using content()
+content(about_req)
+```
+
+```{r}
+# Finally, we will print the display the message and the description
```
Comment (DeenaBleich) [inline Notebooks/How_to_use_ISB-CGC_APIs.Rmd]: How about "Finally, we will print the message and the description"

File: Notebooks/How_to_use_ISB-CGC_APIs.Rmd [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+That wasn't difficult at all! Next we will cover a few of the other information APIs.
+
+## Example: `/data/availabile` Endpoint
+
+The `/data/availabile` Endpoint is designed to return the data sets and programs available on the WebApp along with the projects or studies that are within those data sets and programs. This endpoint returns a more complicated response object which can be accessed as if it was a combination of lists and vectors. We will first retrieve the request and then view if there was an error code within the response.
+
+```{r}
+# First submit the 'get' request to the API
+programs_req <- GET("https://api-dot-isb-cgc.appspot.com/v4/data/available")
+# Check that the wasn't an error with the request
```
Comment (DeenaBleich) [inline Notebooks/How_to_use_ISB-CGC_APIs.Rmd]: How about "Check that there wasn't..."

File: Notebooks/How_to_use_ISB-CGC_APIs.Rmd [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+
+```{r}
+# Finally, we will print the display the message and the description
+content(about_req)$message
+content(about_req)$documentation
+```
+
+That wasn't difficult at all! Next we will cover a few of the other information APIs.
+
+## Example: `/data/availabile` Endpoint
```
Comment (DeenaBleich) [inline Notebooks/How_to_use_ISB-CGC_APIs.Rmd]: Should it be data/available?

File: Notebooks/How_to_use_ISB-CGC_APIs.Rmd [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+
+The `/data/availabile` Endpoint is designed to return the data sets and programs available on the WebApp along with the projects or studies that are within those data sets and programs. This endpoint returns a more complicated response object which can be accessed as if it was a combination of lists and vectors. We will first retrieve the request and then view if there was an error code within the response.
```
Comment (DeenaBleich) [inline Notebooks/How_to_use_ISB-CGC_APIs.Rmd]: Should it be data/available?

File: Notebooks/How_to_use_ISB-CGC_APIs.Rmd [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+This last section will cover the get `cohorts` Endpoint which requires authorization before submitting the request to the API. This endpoint retrieves information about user generated cohorts from within the WebApp or with the ISB-CGC API. 
+
+### Notes on Authorization and Credentials
+In order to use several of the ISB-CGC APIs, you need have authorization with ISB-CGC.
+
+The following steps are required to use an API that Requires Authorization:
+
+1. Create Google Cloud Project set up*
+2. Register with the ISB-CGC WebApp*
+3. Create a Creditial File on your local machine by using the [isb_auth.py script](https://github.com/isb-cgc/ISB-CGC-API/blob/master/scripts/isb_auth.py) from the [ISB-CGC-API Repository](https://github.com/isb-cgc/ISB-CGC-API)
```
Comment (DeenaBleich) [inline Notebooks/How_to_use_ISB-CGC_APIs.Rmd]: Should it be "Credentials"?

File: Notebooks/How_to_use_ISB-CGC_APIs.Rmd [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+  * This script can be run from the command line or from within python but has to occur on your local machine.
```
Comment (DeenaBleich) [inline Notebooks/How_to_use_ISB-CGC_APIs.Rmd]: Capitalize Python

File: Notebooks/How_to_use_ISB-CGC_APIs.Rmd [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+**Note:** the credentials file will expire after 1 hour and a new one will need to be generated. If a new file is not generated with the isb_auth script, you can delete the original file and try running the script again.
+
+If you are having any issues, you can contact us at feedback@isb-cgc.org
+
+Finally, we can make a `get` request to the `cohorts` ISB-CGC API.
+
+```{r}
+#add_headers(Authorization="bearer <your api key>")
+cohort_req <- GET("https://api-dot-isb-cgc.appspot.com/v4/cohorts", add_headers(Authorization=head))
+# Check that the wasn't an error with the request
```
Comment (DeenaBleich) [inline Notebooks/How_to_use_ISB-CGC_APIs.Rmd]: "Check that there wasn't..."

File: Notebooks/How_to_use_ISB-CGC_APIs.md [hunk 10]
```diff
@@ -0,0 +1,415 @@
+How to Use ISB-CGC APIs
+================
+
```
Comment (DeenaBleich) [inline Notebooks/How_to_use_ISB-CGC_APIs.md]: Please see above section for comments.

File: Notebooks/How_to_use_ISB-CGC_APIs.Rmd [hunk 11]
```diff
@@ -0,0 +1,244 @@
...
+
+Some useful tutorials and quick start guides on APIs are:
+
+* [GDC's Getting Started guide for APIs](https://docs.gdc.cancer.gov/API/Users_Guide/Getting_Started/)
+* [API Integration in Python](https://realpython.com/api-integration-in-python/)
+* [Python API Tutorial: Getting Started with APIs](https://www.dataquest.io/blog/python-api-tutorial/)
+
+### What is SwaggerUI?
+
+[SwaggerUI](https://swagger.io/tools/swagger-ui/) is a user interface that allows users to try out the APIs and view their documentation easiely. A tutorial on how to use the ISB-CGC APIs on the SwaggerUI can be found [here](https://isb-cancer-genomics-cloud.readthedocs.io/en/latest/sections/progapi/progAPI-v4/Programmatic-Demo.html#apis-demo).
```
[the remark this answers is not in the document]
Comment (lhagen-isb) [reply Notebooks/How_to_use_ISB-CGC_APIs.Rmd]: Fixed! Thanks!

File: Notebooks/How_to_use_ISB-CGC_APIs.md [hunk 12]
```diff
@@ -0,0 +1,415 @@
...
+  - [API Integration in
+    Python](https://realpython.com/api-integration-in-python/)
+  - [Python API Tutorial: Getting Started with
+    APIs](https://www.dataquest.io/blog/python-api-tutorial/)
+
+### What is SwaggerUI?
+
+[SwaggerUI](https://swagger.io/tools/swagger-ui/) is a user interface
+that allows users to try out the APIs and view their documentation
+easiely. A tutorial on how to use the ISB-CGC APIs on the SwaggerUI can
```
[the remark this answers is not in the document]
Comment (lhagen-isb) [reply Notebooks/How_to_use_ISB-CGC_APIs.md]: Fixed! Thanks!

File: Notebooks/How_to_use_ISB-CGC_APIs.md [hunk 13]
```diff
@@ -0,0 +1,415 @@
...
+``` r
+# install.packages("httr")
+library(httr)
+```
+
+    ## Warning: package 'httr' was built under R version 3.6.1
+
+## Use cases for ISB-CGC APIs
+
+The ISB-CGC APIs can be used for an number of different tasks for
```
[the remark this answers is not in the document]
Comment (lhagen-isb) [reply Notebooks/How_to_use_ISB-CGC_APIs.md]: Fixed! Thanks!

File: Notebooks/How_to_use_ISB-CGC_APIs.Rmd [hunk 14]
```diff
@@ -0,0 +1,244 @@
...
+## Example: `about` Endpoint
+
+We are first going to explore the `about` endpoint using the 'get' request to the API. This API will give you information about the ISB-CGC API such a link to the Swagger UI interface and the documentation.
+
+```{r}
+# First submit the 'get' request to the API
+about_req <- GET("https://api-dot-isb-cgc.appspot.com/v4/about")
+```
+
+Now that we have the request response, we are going to check that we didn't recieve an error code or if the request was successful. If the request was successful, then the status code will come back as 200 but if something went wrong then the status code may be something 404 or 503. If you have recieved any error codes, you can check out Google's [Troubleshooting response errors guide](https://cloud.google.com/endpoints/docs/openapi/troubleshoot-response-errors).
```
[the remark this answers is not in the document]
Comment (lhagen-isb) [reply Notebooks/How_to_use_ISB-CGC_APIs.Rmd]: Fixed! Thanks!

File: Notebooks/How_to_use_ISB-CGC_APIs.Rmd (hunk 6 above)
Comment (lhagen-isb) [reply Notebooks/How_to_use_ISB-CGC_APIs.Rmd]: Yes... I'm not sure how that typo got in there. Fixed! Thanks!

File: Notebooks/How_to_use_ISB-CGC_APIs.Rmd (hunk 8 above)
Comment (lhagen-isb) [reply Notebooks/How_to_use_ISB-CGC_APIs.Rmd]: Fixed! Thanks!

File: Notebooks/How_to_use_ISB-CGC_APIs.Rmd (hunk 1 above)
Comment (lhagen-isb) [reply Notebooks/How_to_use_ISB-CGC_APIs.Rmd]: Fixed! Thanks!

File: Notebooks/How_to_use_ISB-CGC_APIs.Rmd (hunk 2 above)
Comment (lhagen-isb) [reply Notebooks/How_to_use_ISB-CGC_APIs.Rmd]: Fixed! Thanks!

File: Notebooks/How_to_use_ISB-CGC_APIs.Rmd (hunk 3 above)
Comment (lhagen-isb) [reply Notebooks/How_to_use_ISB-CGC_APIs.Rmd]: Fixed! Thanks!

File: Notebooks/How_to_use_ISB-CGC_APIs.Rmd (hunk 4 above)
Comment (lhagen-isb) [reply Notebooks/How_to_use_ISB-CGC_APIs.Rmd]: Fixed! Thanks!

File: Notebooks/How_to_use_ISB-CGC_APIs.Rmd (hunk 5 above)
Comment (lhagen-isb) [reply Notebooks/How_to_use_ISB-CGC_APIs.Rmd]: Fixed! Thanks!

File: Notebooks/How_to_use_ISB-CGC_APIs.Rmd (hunk 7 above)
Comment (lhagen-isb) [reply Notebooks/How_to_use_ISB-CGC_APIs.Rmd]: Fixed! Thanks!

File: Notebooks/How_to_use_ISB-CGC_APIs.Rmd (hunk 9 above)
Comment (lhagen-isb) [reply Notebooks/How_to_use_ISB-CGC_APIs.Rmd]: Fixed! Thanks!

File: Notebooks/How_to_use_ISB-CGC_APIs.md (hunk 10 above)
Comment (lhagen-isb) [reply Notebooks/How_to_use_ISB-CGC_APIs.md]: Fixed! Thanks!

**Pull request merged**